javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -165,8 +164,8 @@ private Vector(BitMappedTrie<T> trie) {
             final Object[] array = ((Collection<? extends T>) iterable).toArray();
             return ofAll(BitMappedTrie.ofAll(array));
         } else {
-            final Seq<? extends T> elems = seq(iterable);
-            final Object[] array = asArray(elems.iterator(), elems.size());
+            final Seq<? extends T> result = (iterable instanceof Seq) ? (Seq<T>) iterable : List.ofAll(iterable);
+            final Object[] array = asArray(result.iterator(), result.size());
```

Comment:
Right - but do it. The problem of consumption also occurs in other places, e.g. when an TraversableOnce is used in a For loop etc. It is a design-error of the type hierarchy that will be fixed with 3.0.